### PR TITLE
chore: Rename BulkVotingStrategy -> QuadraticFundingVotingStrategy, add coverage for QuadraticFundingVotingStrategy

### LINKED ISSUE #359: Rename BulkVote -> QuadraticFundingVote
**User Story:**

- Rename BulkVote contract to `QuadraticFundingVote`
- Get test coverage of BulkVote to 100%

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### PR DESCRIPTION
**Pull request by t-proctor**
##### Description

- [x] Refactor all files with `QuadraticFundingVotingStrategy` instead of `BulkVotingStrategy`
- [x] Run smart contracts locally 
- [x] Add full coverage to `quadraticFundingVotingStrategy.test.ts`
- [x] Deploy/verify contract on Goerli
- [ ] Deploy/verify contract on Optimism

##### Refers/Fixes

fixes #359 

##### Testing

**Note** 
This story is very close to finished. The only thing I think that is needed is adding a working grant address to `quadraticFundingVotingStrategy.test.ts` (to make test pass) and Optimism deploy

### CODE REVIEW

File: packages/contracts/docs/CHAINS.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ This document lists all the addresses of the contracts that have been deployed o
 
 | Contract              | Address                                    |
 |-----------------------|--------------------------------------------|
-| BulkVotingStrategy    | 0x9d55810da86A9610Ab3223f4CDF0b5d81FfA579c |
+| QuadraticFundingVotingStrategy    | 0x9d55810da86A9610Ab3223f4CDF0b5d81FfA579c |
```
Comment (thelostone-mc) [inline packages/contracts/docs/CHAINS.md:35]: small nitpick but could you update the formatting of the table using something like https://www.tablesgenerator.com/markdown_tables

**Comment by t-proctor** [comment]
@thelostone-mc do you know why the build is failing? Can't see the details on Fleek

**Comment by t-proctor** [comment]
Also, had some questions @thelostone-mc 
1. What tests would you recommend to make sure the refactor works? So far have only compiled/testing contracts locally. A lot of the refactor seems to be testnet deploy related. Does this need to be tested?
2. I’m adding tests to the QuadraticFundingVotingStrategy. I want to test the event emission, which requires the following: 
`_token.code.length > 0 && success && (returndata.length == 0 || abi.decode(returndata, (bool))))
`
I’m struggling to pass in a _token parameter that is a contract instead of an EOA to fulfill the first requirement. Feel like there’s an obvious way to do this that I am overlooking. My attempt so far is shown in `packages/contracts/test/QuadraticFundingVotingStrategy.test.ts`

I'm free to hop on a call whenever if it would be easier that way for you to answer these questions. Thanks for your time!

**Comment by thelostone-mc** [comment]
> Do you know why the build is failing? Can't see the details on Fleek

This seems to be on-off failure. Re-running the ci job

**Comment by thelostone-mc** [comment]
> What tests would you recommend to make sure the refactor works? So far have only compiled/testing contracts locally. A lot of the refactor seems to be testnet deploy related. Does this need to be tested?

Yes I would encourage you to deploy the contracts and make sure it works as expected

**Comment by t-proctor** [comment]
> > What tests would you recommend to make sure the refactor works? So far have only compiled/testing contracts locally. A lot of the refactor seems to be testnet deploy related. Does this need to be tested?
> 
> Yes I would encourage you to deploy the contracts and make sure it works as expected
> 
> > I’m adding tests to the QuadraticFundingVotingStrategy. I want to test the event emission, which requires the following:
> 
> Hmm one way we could go about doing this is is within `contracts/contracts/test` create a MockToken which implements ERC20 token
> 
> ```
> import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
> 
> contract MockToken is ERC20 {
>   constructor(uint256 initialSupply) ERC20("Mock", "MCK") {
>     _mint(msg.sender, initialSupply);
>   }
> 
>   function mint(uint256 amount) external {
>     _mint(msg.sender, amount);
>   }
> }
> ```
> 
> Then you can deploy this in the test and voila you have a ERC 20 token to work with

Thanks, will work on this!

**Comment by t-proctor** [comment]
> @t-proctor we are coming up on a deadline for the team to implement this ! Would you be able to wrap this by monday/tuesday ? If not -> I'll pull this back into our backlog and have one the core dev take a stab at it

Sorry for taking a while. Will work on it this weekend and lyk by Sunday night if I cannot finish it

**Comment by t-proctor** [comment]
@thelostone-mc this story is very close to being finished. I was unsure about two things: adding a working grant address to quadraticFundingVotingStrategy.test.ts and the Optimism mainnet deploy (does the team require this be only be done from a specific address?). If you have any suggestions or think it would be easier for you/someone else to finish them out, I'm fine with either. Should have some time tomorrow.

**Comment by thelostone-mc** [comment]
> adding a working grant address to quadraticFundingVotingStrategy.test.ts

Hmm a grant address could be use a random address. 

> Optimism mainnet deploy

This i'll take care of it. You've got to pay mainnet fees and it doesn't make sense for you to deploy it on our behalf.
I'll schedule that change when we have wrapped up our contract changes

File: packages/round-manager/src/features/round/RoundApplicationForm.tsx [hunk 2]
```diff
@@ -124,7 +124,7 @@ export function RoundApplicationForm(props: {
 
       const round = {
         ...data,
-        votingStrategy: "0xc76Ea06e2BC6476178e40E2B40bf5C6Bf3c40EF6", // BulkVotingStrategy contract
+        votingStrategy: "0xc76Ea06e2BC6476178e40E2B40bf5C6Bf3c40EF6", // QuadraticundingVotingStrategy contract
```
Comment (thelostone-mc) [inline packages/round-manager/src/features/round/RoundApplicationForm.tsx:127]:
```suggestion
        votingStrategy: "0xc76Ea06e2BC6476178e40E2B40bf5C6Bf3c40EF6", // QuadraticFundingVotingStrategy contract
```

File: packages/contracts/docs/CHAINS.md [hunk 3]
```diff
@@ -4,25 +4,27 @@ This document lists all the addresses of the contracts that have been deployed o
...
-| BulkVotingStrategy    | 0x4a850F463D1C4842937c5Bc9540dBc803D744c9F |
-| RoundFactory          | 0x64ab6F2E11dF8B3Be5c8838eDe3951AC928daE9C |
-| RoundImplementation   | 0xdf25423c9ec15347197Aa5D3a41c2ebE27587D59 |
-| Round                 | 0x2DF6c42dd2d7a13c19Ca5f7858fB7cC05A2933ed |
-| ProgramFactory        | 0xB3Ee4800c93cBec7eD2a31050161240e4663Ff5E |
-| ProgramImplementation | 0xED42e0f4391Fa24E579B16191F6Eb41f934c3B1c |
-| Program               | 0x378cEB3dEb7a80ec1579bfd61EE1EFB76Fc63025 |
+| Contract                          | Address                                    |
+|-----------------------------------|--------------------------------------------|
+| QuadraticFundingVotingStrategy    | 0x4a850F463D1C4842937c5Bc9540dBc803D744c9F |
```
Comment (thelostone-mc) [inline packages/contracts/docs/CHAINS.md:22]:
```suggestion
| QuadraticFundingVotingStrategy    | ----------------------TODO--------------------------- |
```

**Comment by thelostone-mc** [comment]
@t-proctor I've updated the tests to give examples on how we could further break it down

**Comment by t-proctor** [comment]
> @t-proctor I've updated the tests to give examples on how we could further break it down Could you
> 
> * make sure CI is passing
> * deploy the contract one last time on Goerli
> * squash the commits

Ok, will work on it tomorrow. 

Also, idk why it's saying I force-pushed branch from 17d5b53 to 2e33f10. I did a hard reset to 4cc9ca0 and force pushed to try to fix it, but it's not showing up in the UI. Do you have a solution? Sorry for messing it up

**Comment by t-proctor** [comment]
> @t-proctor I've updated the tests to give examples on how we could further break it down Could you
> 
> * make sure CI is passing
> * deploy the contract one last time on Goerli
> * squash the commits

Ok, will work on it tomorrow. 

Also, idk why it's saying I force-pushed branch from 17d5b53 to 2e33f10. I did a hard reset to 4cc9ca0 and force pushed to try to fix it, but it's not showing up in the UI. Do you have a solution? Sorry for messing it up

**Comment by t-proctor** [comment]
@thelostone-mc something came up in my personal life so I cannot close this out today

**Comment by thelostone-mc** [comment]
Contract is deployed : 0xfc3Dd3847Fdc052001c6cB0e60A3A921C44C50F3

**Review by thelostone-mc — APPROVED** [review]

**Review by nfrgosselin — APPROVED** [review]
deferring to eng approval

**Pull request merged**